### PR TITLE
Fix temporary directory removal on Windows

### PR DESCRIPTION
**Pull request by pawamoy**
Issue #345

Draft to see how CI behaves.

### CODE REVIEW

**Comment by Yajo** [comment]
This may be unrelated, but you might be interested in this patch I had to add in the refactor: https://github.com/copier-org/copier/blob/b3190c2abffb04d5f4b21fa71a51bf4036baad5a/copier/main.py#L285-L289

See https://bugs.python.org/issue43095 for more details.

**Comment by pawamoy** [comment]
I don't think the failure on MacOS and Python 3.7 is related to my changes here. I'll add tests anyway, of course.

**Comment by Yajo** [comment]
Ah... rebase please, that's fixed on master.

**Comment by pawamoy** [comment]
The Poetry issue comes from its new core version. The solution is to make it less verbose: `poetry install -v` instead of `-vvv`.

**Comment by pawamoy** [comment]
I'll wait for you to merge #364 and I'll rebase.

**Comment by pawamoy** [comment]
Alright!

I've decomposed the process:

- [x] https://github.com/copier-org/copier/pull/358/commits/9c86e1c341f85f907431f52f1397ba7403ce25c6 adds tests for this particular issue, they pass on Windows for Python 3.8 and 3.9.
- [x] https://github.com/copier-org/copier/pull/358/commits/83915961f2f01d3306ed0855e2b89417f8048475 re-enables Windows for Python 3.6 and 3.7, tests are failing as expected
- [x] https://github.com/copier-org/copier/pull/358/commits/ce00fd7a1253908de6ad9732cf9adc5b21bafbc2 fixes the issue by implementing a more robust deletion in a temporary directory class, tests are passing again on Windows for all tested Python versions

Please **squash** as I pushed commits disabling/enabling things in CI that were only used to speed up CI.

**Comment by pawamoy** [comment]
Pexpect timeout again on MacOs + Python 3.6 :confused:

File: copier/tools.py [hunk 1]
```diff
@@ -135,3 +140,27 @@ def force_str_end(original_str: str, end: str = "\n") -> str:
     if not original_str.endswith(end):
         return original_str + end
     return original_str
+
+
+def handle_remove_readonly(func, path, exc):
```
Comment (Yajo) [inline copier/tools.py:145]: Please add docs.

File: copier/tools.py [hunk 2]
```diff
@@ -135,3 +140,27 @@ def force_str_end(original_str: str, end: str = "\n") -> str:
...
+    excvalue = exc[1]
+    if func in (os.rmdir, os.remove, os.unlink) and excvalue.errno == errno.EACCES:
+        os.chmod(path, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)  # 0777
+        func(path)
+    else:
+        raise exc
+
+
+class TemporaryDirectory(tempfile.TemporaryDirectory):
```
Comment (Yajo) [inline copier/tools.py:154]: Also here

**Review by Yajo — CHANGES_REQUESTED** [review]
Thanks!

File: copier/tools.py (hunk 1 above)
Comment (pawamoy) [reply copier/tools.py]: Done.

File: copier/tools.py (hunk 2 above)
Comment (pawamoy) [reply copier/tools.py:164]: Done.

**Comment by pawamoy** [comment]
Could you re-run the jobs to see if the codecov error goes away? Unless you already did?

**Comment by Yajo** [comment]
Oh man what's up with that mac? 🤦🏼‍♂️ 

I could debug that if somebody buys me a mac. Otherwise I suggest you to mark that test as not forced xfail. 🤷🏼‍♂️

**Comment by Yajo** [comment]
Finally! thanks 😊

**Pull request merged**